flink-tests/src/test/java/org/apache/flink/test/streaming/runtime/TimestampITCase.java

```diff
@@ -159,6 +160,28 @@ public void testWatermarkPropagation() throws Exception {
         }
     }
 
+    @Test
+    public void testSelfUnionWatermark() throws Exception {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setParallelism(1);
+        DataStream<Integer> dataStream1 = env.fromElements(1, 2, 3).setParallelism(1);
+
+        dataStream1
+                .union(dataStream1)
+                .addSink(
+                        new DiscardingSink<Integer>() {
+                            @Override
+                            public void writeWatermark(
+                                    org.apache.flink.api.common.eventtime.Watermark watermark)
+                                    throws Exception {
+                                System.out.println(watermark);
+                            }
+                        });
+        // or  use .addSink(new DiscardingSink<>()); and debug to check that the writeWatermark() on
```

Comment:
left over? or was left for future reference?

Reply:
Yes, I noticed it just after publishing the PR and it's already fixed.